test/ui/pages/common/PageTree.test.js

```diff
@@ -36,155 +36,152 @@ const props = {
   onClickClone: jest.fn(),
   onClickPublish: jest.fn(),
   onClickUnPublish: jest.fn(),
+  onClickViewPublishedPage: jest.fn(),
+  onClickPreview: jest.fn(),
+  domain: '',
+  loading: false,
+  pages: PAGES,
+  locale: 'en',
 };
 
+jest.unmock('react-redux');
+
+const requiredState = {
+  modal: { info: {}, visibleModal: '' },
+  pages: { map: {} },
+};
+
+const renderComponent = (addProps = {}) => render(
+  mockRenderWithIntlDDStoreRouter(
+    <PageTree {...props} {...addProps} />,
+    requiredState,
+  ),
+);
+
 describe('PageTree', () => {
   beforeEach(jest.clearAllMocks);
+  afterEach(cleanup);
 
   describe('basic rendering', () => {
-    let component;
-    beforeEach(() => {
-      component = shallow(<PageTree pages={PAGES} locale="en" {...props} />);
-    });
+    beforeEach(renderComponent);
+
     it('renders without crashing', () => {
-      expect(component.exists()).toBe(true);
+      expect(screen.getByRole('table')).toBeInTheDocument();
     });
-    it('renders a table row for each page', () => {
-      expect(component.find('.PageTree__row')).toHaveLength(PAGES.length);
+    it('renders pages', () => {
+      const resultSet = within(screen.queryAllByRole('rowgroup')[1]);
+      const rows = resultSet.getAllByRole('row');
+      expect(rows).toHaveLength(PAGES.length);
     });
   });
 
   describe('drag and drop', () => {
-    let component;
     const handleDropPage = jest.fn();
-    beforeEach(() => {
-      component = shallow((
-        <PageTree
-          pages={PAGES}
-          onDropPage={handleDropPage}
-          {...props}
-        />));
-    });
-    it('calls onDropPage with action of "drop into" if a row is dropped with drop type "medium"', () => {
-      component.instance().handleDrop(DDTable.DROP_MEDIUM, PAGES[1], PAGES[0]);
+
+    it('calls onDropPage with action of "dropping"', () => {
+      renderComponent({ onDropPage: handleDropPage });
+      const firstItem = screen.getByText(PAGES[0].title).closest('tr');
+      const secondItem = screen.getByText(PAGES[1].title).closest('tr');
+      const firstItemHandle = firstItem.querySelector('button');
+      fireEvent.dragStart(firstItemHandle);
+      fireEvent.dragOver(secondItem);
+      fireEvent.drop(secondItem);
       expect(handleDropPage).toHaveBeenCalledWith(
-        PAGES[1].code, PAGES[0].code,
+        PAGES[0].code, PAGES[1].code,
         MovePageModalContainer.INTO_PARENT,
```

Comment:
Can we use a different object with the move positions instead of adding them to the Container?